next-app/src/app/(app-layout)/flavours/[flavourId]/page.tsx

```diff
@@ -19,18 +28,20 @@ export async function generateMetadata({ params }: FlavourPageProps): Promise<Me
     title: flavour.name,
     pathname: `/flavours/${params.flavourId}`,
   });
+  
 }
 
 export default async function FlavourPage({ params }: FlavourPageProps) {
   const flavour = await getObjectById<Flavour>(Number(params.flavourId), dataTypes.FLAVOURS);
-
+  const whiskey = await getObjectById<Whiskey>(Number(params.whiskeyId), dataTypes.WHISKEYS);
```

Comment:
Instead of retreiving a whiskey, then doing "most like by flavour/chemical" etc.

You need some logic that does the following:
- Get the ID of the Target Flavour / Chemical
- Get all Whiskeys
- Sort the Whiskeys (Highest to Lowest) by the largest value in the whiskey.chemicals[ID] / whiskey.flavours[ID], then return top 5~ slice.

Whiskey.flavours = [0,1,2,3,4,5,6,7,8,9] (note numbers are note 0->9, but just as example)

If Id of Flavour = 1, then sort by largest to smallest whiskey.flavours[1] and return .slice(5)

Reply:
Can u point that the code which do i can reference to fix that code?